src/Network/Datadog/Internal.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Network.Datadog.Internal
+( datadogHttp
+, decodeDatadog
+) where
+
+
+import Control.Exception
+
+import Data.Aeson
+import qualified Data.ByteString as BS (ByteString)
+import qualified Data.ByteString.Lazy as LBS (ByteString, empty)
+import Data.Maybe
+import Data.Text (pack)
+import Data.Text.Lazy (unpack)
+import Data.Text.Encoding (encodeUtf8)
+import Data.Text.Lazy.Encoding (decodeUtf8)
+
+import Network.HTTP.Conduit
+
+import Network.Datadog
+
+
+datadogHttp :: Environment-> String -> [(String, String)] -> BS.ByteString -> Maybe LBS.ByteString -> IO LBS.ByteString
+datadogHttp (Environment keys manager) endpoint query httpMethod content = do
+  initReq <- parseUrl $ "https://app.datadoghq.com/api/v1/" ++ endpoint
```

Comment:
There's an enterprise version of datadog too, I believe. Might be good to make the base URI configurable.